src/RecordSession.cc

```diff
@@ -228,6 +228,13 @@ static void note_entering_syscall(RecordTask* t) {
      * (if it's not a SYS_restart_syscall restart)
      * will use the original registers. */
     t->ev().Syscall().regs = t->regs();
+  } else {
+    t->ev().Syscall().regs.set_syscallno(t->regs().syscallno());
```

Comment:
Why do we need to do this?

Reply:
x8 gets set to restart_syscall after the end of the signal stop, but the EV_SYSCALL_INTERRUPTION still has the x8 from the original syscall entry. In the EV_SIGNAL_DELIVERY case below, we fix that up for that event, but we never fix it up for the syscall event itself. I suppose we could just change it there, and arg1 as well, but in the next commit, we also need to fix x7 here due to the kernel bug, so I think it makes sense to keep those changes together.

Reply:
(x8 is syscallno)